### ISSUE TITLE
Get preferences object on extension __init__ method

### ISSUE THREAD

**Issue by brpaz** [Ulauncher/Ulauncher#244]
Hello.
I am building an extension where I need to get access to the extension preferences in the "init" method of the Extension.
But the preferences object is always empty there. I understood that the preferences are initialized later and  the "PreferencesEvent" is triggered when that happens. But I couldn't find a way to listen to that event in my extension, only the "PreferencesUpdatedEvent" which is called only when the user updates a value in the preferences window.

Basically, my scenario is, I am instantiating a client object for an API in my extension __init__ method and I need to get an "access token" stored in the preferences. A workaround is that I set it everytime on the "KeywordQueryEventListener". I can do that, but I think it should be a better way.

Thank you and keep up the good work. Ulauncher is amazing ;)

**Comment by brpaz** [comment]
well, I discovered the search plugin does what I want and I manage to get it working. Dont know why it wasn't working before

**Comment by gornostal** [comment]
Apologies for the delay. I checked the code and I don't see a way to get preferences in the `__init__`.
The earliest you can get preferences is with `PreferencesEvent`
